server/handlers.go

```diff
@@ -0,0 +1,209 @@
+	versionByte, err := parseVersionByte(w, r)
+	if err != nil {
+		return fmt.Errorf("error parsing version byte: %w", err)
+	}
+	commitmentMeta := commitments.CommitmentMeta{
+		Mode:        commitments.OptimismGeneric,
+		CertVersion: versionByte,
+	}
+
+	rawCommitmentHex, ok := mux.Vars(r)["raw_commitment_hex"]
+	if !ok {
+		return fmt.Errorf("commitment not found in path: %s", r.URL.Path)
+	}
+	commitment, err := hex.DecodeString(rawCommitmentHex)
+	if err != nil {
+		return fmt.Errorf("failed to decode commitment %s: %w", rawCommitmentHex, err)
+	}
+
+	svr.log.Info("Processing op keccak commitment", "commitment", rawCommitmentHex, "commitmentMeta", commitmentMeta)
+	return svr.handleGetShared(r.Context(), w, commitment, commitmentMeta)
+}
+
+func (svr *Server) handleGetShared(ctx context.Context, w http.ResponseWriter, comm []byte, meta commitments.CommitmentMeta) error {
+	input, err := svr.router.Get(ctx, comm, meta.Mode)
+	if err != nil {
+		err = MetaError{
+			Err:  fmt.Errorf("get request failed with commitment %v: %w", comm, err),
+			Meta: meta,
+		}
+		if errors.Is(err, ErrNotFound) {
+			http.Error(w, err.Error(), http.StatusNotFound)
+		} else {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		return err
+	}
+
+	svr.writeResponse(w, input)
+	return nil
+}
+
+// =================================================================================================
+// PUT ROUTES
+// =================================================================================================
+
+func (svr *Server) handlePutSimpleCommitment(w http.ResponseWriter, r *http.Request) error {
+	svr.log.Info("Processing simple commitment")
+	commitmentMeta := commitments.CommitmentMeta{
+		Mode:        commitments.SimpleCommitmentMode,
+		CertVersion: byte(commitments.CertV0), // TODO: hardcoded for now
+	}
+	return svr.handlePutShared(w, r, nil, commitmentMeta)
+}
+
+// handleGetOPKeccakCommitment handles the GET request for optimism keccak commitments.
+func (svr *Server) handlePutOPKeccakCommitment(w http.ResponseWriter, r *http.Request) error {
+	// TODO: do we use a version byte in OPKeccak commitments? README seems to say so, but server_test didn't
+	// versionByte, err := parseVersionByte(r)
+	// if err != nil {
+	// 	err = fmt.Errorf("error parsing version byte: %w", err)
+	// 	http.Error(w, err.Error(), http.StatusBadRequest)
+	// 	return err
+	// }
+	commitmentMeta := commitments.CommitmentMeta{
+		Mode:        commitments.OptimismKeccak,
+		CertVersion: byte(commitments.CertV0),
+	}
+
+	rawCommitmentHex, ok := mux.Vars(r)["raw_commitment_hex"]
+	if !ok {
+		return fmt.Errorf("commitment not found in path: %s", r.URL.Path)
+	}
+	commitment, err := hex.DecodeString(rawCommitmentHex)
+	if err != nil {
+		return fmt.Errorf("failed to decode commitment %s: %w", rawCommitmentHex, err)
+	}
+
+	svr.log.Info("Processing op keccak commitment", "commitment", rawCommitmentHex, "commitmentMeta", commitmentMeta)
```

Comment:
we use this log for both dispersal and retrieval - should we add a method type to the string to allow for observable differentiation?

Reply:
[a36001e](https://github.com/Layr-Labs/eigenda-proxy/pull/185/commits/a36001e07f71ea74b1323339cb8e84a34f7fcd22)